ykrt/src/mt.rs

```diff
@@ -283,25 +295,40 @@ impl MTThread {
                     };
 
                     if stop_tracer {
-                        if pack.compare_and_swap(lp, PHASE_COMPILED, Ordering::Release) == lp {
-                            let _sir_trace = Rc::get_mut(&mut self.inner)
-                                .unwrap()
-                                .tracer
-                                .take()
-                                .unwrap()
-                                .0
-                                .stop_tracing();
-                            // FIXME build TIR and compile. Eventually in a background thread?
+                        let sir_trace = Rc::get_mut(&mut self.inner)
+                            .unwrap()
+                            .tracer
+                            .take()
+                            .unwrap()
+                            .0
+                            .stop_tracing()
+                            .unwrap();
+                        let tir_trace = TirTrace::new(&*SIR, &*sir_trace).unwrap();
```

Comment:
I think we should add a comment with a few words explaining what is going on here. About avoiding drops etc.

Reply:
Fixed in 789a136.